.github/workflows/testsuite.yml

```diff
@@ -0,0 +1,398 @@
+  # \__ \ '  \/ _ \ / / -_)  | || ' \|  _/ _ \ '_| '  \/ _` |  _| / _ \ ' \(_-<
+  # |___/_|_|_\___/_\_\___| |___|_||_|_| \___/_| |_|_|_\__,_|\__|_\___/_||_/__/
+
+  authors:
+    runs-on: ubuntu-latest
+    # not from a fork and base_ref defined
+    if: github.event.pull_request.head.repo.full_name == github.repository && github.base_ref != ''
+    continue-on-error: true
+
+    steps:
+      - name: Dump GitHub context
+        env:
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+        run: echo "$GITHUB_CONTEXT"
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 1000
+      - name: Involved authors
+        run: |
+          echo "Pull request authors"
+          echo "# git merge-base origin/${BASE_REF} HEAD"
+          branch_off=$(git merge-base origin/${BASE_REF} HEAD)
+          git log --pretty=format:"Author: %an <%ae>" $branch_off..${SHA}^2 | sort -u
+        env:
+          BASE_REF: ${{ github.base_ref }}
+          SHA: ${{ github.sha }}
+
+  #  ___            _ _           ___ _           _
+  # / __| __ _ _ _ (_) |_ _  _   / __| |_  ___ __| |__
+  # \__ \/ _` | ' \| |  _| || | | (__| ' \/ -_) _| / /
+  # |___/\__,_|_||_|_|\__|\_, |  \___|_||_\___\__|_\_\
+  #                       |__/
+  # ascii art small font from https://ascii.co.uk/text
+
+  sanity_check:
+    name: "Sanity: Linux -Dusethreads"
+    runs-on: ubuntu-latest
+    timeout-minutes: 120
+
+    env:
+      PERL_SKIP_TTY_TEST: 1
+      CONTINUOUS_INTEGRATION: 1
+      WORKSPACE: ${{ github.workspace }}
+
+    steps:
+      - name: Dump GitHub context
+        env:
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+        run: echo "$GITHUB_CONTEXT"
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: git cfg + fetch tags
+        run: |
+          git config diff.renameLimit 999999
+          git fetch --depth=1 origin +refs/tags/*:refs/tags/*
+      - name: Configure
+        run: |
+          ./Configure -des -Dusedevel ${CONFIGURE_ARGS} -Dprefix="$HOME/perl-blead" -DDEBUGGING
+        env:
+          CONFIGURE_ARGS: "-Dusethreads"
+      - name: Build
+        run: |
+          make -j2
+      - name: Run Tests
+        run: |
+          TEST_JOBS=2 make -j2 test
+
+  # use a dedicated job to check if full testsuite is enabled
+  #   avoid starting VMs if not neeced
+  #   and also provide the same check for all platforms
+  # the job fails when ENABLE_FULL_TEST_SUITE is unset
+  # alternatively could upload an artifact file and abort...
+
+  check_full_test_suite:
+    name: "ENABLE_FULL_TEST_SUITE set?"
+    runs-on: ubuntu-latest
+
+    steps:
+      # --------------------------------------------------------------------------------------------
+      # cannot use secrets earlier as part of 'if:' condition
+      - name: "Runs if ENABLE_FULL_TEST_SUITE is set"
+        run: |
+          if [ -z "${ENABLE_FULL_TEST_SUITE}" ]; then
```

Comment:
need to understand why ENABLE_FULL_TEST_SUITE is not set here

Reply:
Was this resolved?

Reply:
no this is not solved, this is probably a limitation of secrets, I need to investigate the best way to integrate them in a shell script
I wonder if they are not obfuscated

Reply:
You wrote recently that you think it's ready. Does it mean you resolved this issue?